engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -676,6 +676,31 @@ yield switch (identifier) {
               yield new IR$Function$Lambda(args, body, getIdentifiedLocation(tree), true, meta(), diag());
             }
           }
+          case "=" -> {
```

Comment:
@kazcw , it is slightly unfortunate I have to recognize `=` as function definition. Might be easier to have `Tree.Function` at such places.

Reply:
If it doesn't parse as a `Function`, it shouldn't be reinterpreted as a function in `TreeToIr`. The problem with this test case has nothing to do with functions: Higher in the tree, the body of the `else` branch parsed as an `ArgumentBlockApplication`, which it clearly is not. The `if`/`else` macro needs to translate an `else` block into a `BodyBlock`, so that its contents are interpreted in statement context, where `Function` definitions can occur. Thank you for the test case; i'll fix it and this code will not be necessary.

Reply:
Great. I'll merge tomorrow, if the fix is ready

Reply:
Fixed in #3785.